File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/LocatorStateEventIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class LocatorStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'locatorId' => 'string',
            'version' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var LocatorStateEventId
     */
    private $value;

    /**
     * @param LocatorStateEventId $value
     */
    public function __construct(LocatorStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new LocatorStateEventId();
        }
    }

    /**
     * @return LocatorStateEventId
     */
    public function toLocatorStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getLocatorId()
    {
        return $this->value->getLocatorId();
    }

    /**
     * @param string $locatorId
     */
    public function setLocatorId($locatorId)
    {
        $this->value->setLocatorId($locatorId);
    }

    /**
     * @return Long
     */
    public function getVersion()
    {
        return $this->value->getVersion();
    }

    /**
     * @param Long $version
     */
    public function setVersion($version)
    {
        $this->value->setVersion($version);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getLocatorId(),
            $this->getVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return LocatorStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setLocatorId($pValues[0]);
        $this->setVersion($pValues[1]);
        return $this;
    }

}

File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/LocatorStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class LocatorStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "locatorId",
            "version",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private LocatorStateEventId value;

    public LocatorStateEventIdFlattenedDto()
    {
        this(new LocatorStateEventId());
    }

    public LocatorStateEventIdFlattenedDto(LocatorStateEventId value)
    {
        this.value = value;
    }

    public LocatorStateEventId toLocatorStateEventId()
    {
        return this.value;
    }


    public String getLocatorId()
    {
        return this.value.getLocatorId();
    }

    public void setLocatorId(String locatorId)
    {
        this.value.setLocatorId(locatorId);
    }

    public Long getVersion()
    {
        return this.value.getVersion();
    }

    public void setVersion(Long version)
    {
        this.value.setVersion(version);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        LocatorStateEventIdFlattenedDto other = (LocatorStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

